server/item/enchantment/silk_touch.go

```diff
@@ -0,0 +1,33 @@
+package enchantment
+
+import "github.com/df-mc/dragonfly/server/item"
+
+// SilkTouch is an enchantment that allows many blocks to drop themselves instead of their usual items when mined.
+type SilkTouch struct{ enchantment }
+
+// Name ...
+func (e SilkTouch) Name() string {
+	return "Silk Touch"
+}
+
+// MaxLevel ...
+func (e SilkTouch) MaxLevel() int {
+	return 1
+}
+
+// WithLevel ...
+func (e SilkTouch) WithLevel(level int) item.Enchantment {
+	return SilkTouch{e.withLevel(level, e)}
+}
+
+// CompatibleWith ...
+func (e SilkTouch) CompatibleWith(s item.Stack) bool {
+	it := s.Item()
+	_, pickaxe := it.(item.Pickaxe)
+	_, axe := it.(item.Axe)
+	_, shovel := it.(item.Shovel)
+	_, hoe := it.(item.Hoe)
+	_, shears := it.(item.Shears)
```

Comment:
Might want to simply check if it's a tool.Tool and check if its tool type is one of these 5. That way custom tools can have silk touch too.

Reply:
The protection enchantments should probably be changed to check if they implement the armour interfaces instead of concrete types too.